File: leetcode/1-two-sum/main.go
/*
	Problem: Two Sum
	Link: https://leetcode.com/problems/two-sum/description/
*/

package main

func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int)

	for i, num := range nums {
		expect := target - num

		if idx, found := numMap[expect]; found {
			return []int{idx, i}
		}

		// Store current num & i to numMap
		numMap[num] = i
	}

	return nil
}

File: leetcode/1-two-sum/Main.kt
package two_sum

fun main() {
    val solution = Solution()
    println(solution.twoSum(intArrayOf(2, 7, 11, 15), 9).contentToString())
}

class Solution {
  fun twoSum(nums: IntArray, target: Int): IntArray {
    val diffMap = mutableMapOf<Int, Int>()

    nums.forEachIndexed { index, num -> 
      val expectNum = target - num

      diffMap[expectNum]?.let { return intArrayOf(it, index) }
      diffMap[num] = index
    }
    return intArrayOf()
  }
}